tests/test_setup_kubernetes_job.py

```diff
@@ -391,3 +391,31 @@ def test_setup_kube_deployments_rate_limit():
             rate_limit=0,
         )
         assert fake_app.create.call_count == 3
+
+
+def test_setup_kube_deployments_skip_malformed_apps():
+    with mock.patch(
+        "paasta_tools.setup_kubernetes_job.create_application_object", autospec=True,
+    ) as mock_create_application_object, mock.patch(
+        "paasta_tools.setup_kubernetes_job.list_all_deployments", autospec=True
+    ), mock.patch(
+        "paasta_tools.setup_kubernetes_job.log", autospec=True
+    ) as mock_log_obj:
+        mock_client = mock.Mock()
+        mock_service_instances = ["fake.instance"]
+        fake_app = mock.Mock(create=mock.Mock())
+        fake_app.create = mock.Mock(side_effect=Exception("Kaboom!"))
+        fake_app.__str__ = mock.Mock(return_value="fake_app")
+        mock_create_application_object.return_value = (True, fake_app)
+
```

Comment:
It looks like this test isn't actually verifying we'd 'skip' one with an exception and move onto the next since we're only testing 1 `mock_service_instances`. We *are* verifying we caught an exception, so that's great, but we should also do as the test is named and verify w/ multiple `mock_service_instances`, making sure we caught the exception on the first but can successfully move on to the 2nd. 

You should be able to configure multiple side effects for `mock_create_application_object`, making sure we actually called the 2nd app's `.create` after catching the exception on the first one.